base/thread_pool.h

```diff
@@ -17,15 +16,19 @@ namespace slinky {
 // It is not directly used by anything except for testing.
 class thread_pool {
 public:
+  using task_id = const void*;
```

Comment:
style nit: might be better to do something like `struct task_id_impl {}; using task_id = const task_id_impl*;` so that no other pointers can implicitly convert to a `task_id`?

Reply:
I think the idea I was going for was that people would in fact use arbitrary convenient objects as task ids. In the case of `parallel_for`, it uses the state object it creates.